players/computer_player.py

```diff
@@ -0,0 +1,158 @@
+            elif c is Color.YELLOW:
+                self.grey_chars[i] += guess[i]
+                self.yellow_chars += guess[i]
+
+    def __eliminate_words(self, guess, colors):
+        rx = r""
+
+        # Build regex (for eliminating impossible words)
+        for i, color in enumerate(colors):
+            if color is Color.GREEN:  # i'th letter is known
+                rx += guess[i]
+            else:
+                rx += EXCLUSION_RX.format(chars=self.grey_chars[i])
+
+        # Filter function
+        regex_filter = lambda word: re.search(rx, word)
+        yellow_filter = lambda word: self.__contains_all_yellow_chars(word)
+        both_filter = lambda word: (regex_filter(word) and yellow_filter(word))
+
+        # Eliminate impossible words
+        self.possible_words[::] = list(filter(both_filter, self.possible_words))
+
+    def __calc_next_guess(self):
+        """
+        The logic of choosing a guess.
+        First we calculate chars frequencies so we can value each word. The more
+        frequent a word's chars the higher her score. Eventually we choose the
+        word with the highest value as our next guess.
+        That way we can try to eliminate as many words as possible for our next
+        guess.
+
+        :return: string, the guess that should be applied this turn.
+        """
+        chars_frequencies = self.__calc_chars_frequency(self.possible_words)
+
+        word_value_fn = lambda word: self.__word_value(word, chars_frequencies, self.yellow_chars)
+        self.possible_words.sort(key=word_value_fn)
+        if self.possible_words:
+            return self.possible_words.pop()
+        else:
+            raise ValueError("Words list is empty")
+
+    def __contains_all_yellow_chars(self, word):
+        """
+        Tells if given word contains all yellow chars.
+
+        :param word: tested word
+        :return: Bool
+        """
+        yellow_chars_list = list(self.yellow_chars)  # Stack
+        word_copy = word
+
+        while yellow_chars_list:
+            char = yellow_chars_list.pop()
+            if char not in word_copy:
+                return False
+            word_copy.replace(char, "", 1)
+
+        return True
+
+    @staticmethod
+    def __calc_max_unique_chars(words):
+        """
+        Calculate the word with most unique chars in a given words list
+
+        :param words: The list
+        :return: number of max unique chars in a word in words.
+        """
+        result = 0
+
+        if words:
+            str_max_unique_chars = max(words, key=count_unique_chars)
+            result = len(set(str_max_unique_chars))
+
+        return result
+
+    @staticmethod
+    def __calc_chars_frequency(words):
+        frequencies = {char: 0 for char in ALPHABET_LOWER}
+
+        for word in words:
+            for char in word:
+                frequencies[char] += 1
+
+        return frequencies
+
+    @staticmethod
+    def __word_value(word, frequencies, yellow_chars):
```

Comment:
Why are you using dunders everywhere? 🤔

Reply:
To mark method private, only for internal class use

Reply:
That's not what they are intended for 🤓 watch this thingy:
https://www.youtube.com/watch?v=HTLu2DFOdTg&t=33m8s

Reply:
Just use 1 _ everywhere